pkg/ibmcsidriver/controller_helper.go

```diff
@@ -329,8 +339,29 @@ func getVolumeParameters(logger *zap.Logger, req *csi.CreateVolumeRequest, confi
 
 	//TODO port the code from VPC BLOCK to find region if zone is given
 
+	// validate bandwidth for dp2 profile
+	if volume.VPCVolume.Profile != nil && volume.VPCVolume.Profile.Name == DP2Profile && volume.VPCVolume.Bandwidth > 0 {
+		err = fmt.Errorf("bandwidth is not supported for dp2 profile; please remove the property or set it to zero")
+		logger.Error("getVolumeParameters", zap.NamedError("invalidParameter", err))
+		return volume, err
+	}
+
+	// validation zone and iops for 'rfs' profile
+	if volume.VPCVolume.Profile != nil && volume.VPCVolume.Profile.Name == RFSProfile {
```

Comment:
I think these code block can be optimised, we are checking Profile != nil again and again

Reply:
I had given this comment to @renukasingare but seems this is still pending I will rectify this.

Reply:
DOne